src/components/arc-focus-trap.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import {
+  ArcContext,
+  Composable,
+  findFocusable,
+  renderComposed,
+  requireContext,
+} from '../internal-types';
+import { StateContainer } from '../state/state-container';
+
+/**
+ * The ArcFocusTrap prevents focus from leaving the given area of the DOM,
+ * until the trap is destroyed. When initially mounted, it will move
+ * focus to the first focusable element inside of it, or the focusIn element,
+ * if the focus hasn't yet been changed (i.e. if there's no other autofocused
+ * element).
+ *
+ * When unmounted, it will give focus back to the element focused before the
+ * trap was created, unless an override is given
+ *
+ * @example
+ * <FocusTrap><MyContent /></FocusTrap>
+ *
+ * <FocusTrap focusIn=".submit" focusOut=".open-modal-button">
```

Comment:
Should it support focusing React Components as well?

Reply:
Is there a way to pass in a callback as props to children that then pass in selectors to focus trap? It feels like FocusTrap could potentially know too much about it's children and can lead to bugs if classes are ever changed in it's children.